### PR TITLE
[Core] Explicit encoding for opening files and update pylint

### LINKED ISSUE #3152: [Core] Logging raises encoding error
Our code in `log_lib` raises encoding error while writing the logs:
```
UnicodeEncodeError: 'latin-1' codec can't encode characters in position 5-44: ordinal not in range(256)
```

https://github.com/skypilot-org/skypilot/blob/2f0432bff3a70d264f4b2cfde91220c9670c98fa/sky/skylet/log_lib.py#L68-L99

This could happen when the default encoding method is not `utf-8` for the system.

https://chat.openai.com/share/2725e834-0a2d-4edf-8e0c-62f3f8afd915

It is recommended to set `utf-8` for `open` and the behavior will be changed to use `utf-8` by default in `python 3.15` according to PEP:
https://peps.python.org/pep-0597/
https://peps.python.org/pep-0686/

### PR DESCRIPTION
**Pull request by Michaelvll**
Fixes #3152

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

**Review by romilbhardwaj — COMMENTED** [review]
Thanks @Michaelvll! Looks like we need to be careful about all `open()` calls.

As an alternative, can we somehow globally set to use utf-8  for all calls? That reduces chances of bugs if a developer forgets to add `encoding='utf-8'`.

 Looks like the [UTF-8 mode](https://docs.python.org/3/library/os.html#python-utf-8-mode) aims to do so, and can be forced by setting `PYTHONUTF8=1` environment variable. Any way we can use this?

**Comment by Michaelvll** [comment]
> Thanks @Michaelvll! Looks like we need to be careful about all `open()` calls.
> 
> As an alternative, can we somehow globally set to use utf-8 for all calls? That reduces chances of bugs if a developer forgets to add `encoding='utf-8'`.
> 
> Looks like the [UTF-8 mode](https://docs.python.org/3/library/os.html#python-utf-8-mode) aims to do so, and can be forced by setting `PYTHONUTF8=1` [environment variable](https://stackoverflow.com/a/74863886). Any way we can use this?

Thanks for pointing that out @romilbhardwaj! It seems that explicitly specifying the `encoding` for every `open` is a recommended practice according to the PEP standard: https://peps.python.org/pep-0597/ and also a warning in `pylint`: https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/unspecified-encoding.html

It might be a good chance for us to add the encoding everywhere in our code and updating our pylint version. Wdyt?

File: .pylintrc [hunk 1]
```diff
@@ -157,12 +160,6 @@ disable=abstract-method,
 # mypackage.mymodule.MyReporterClass.
 output-format=text
 
-# Put messages in a separate file for each module / package specified on the
-# command line instead of printing them on stdout. Reports (if any) will be
-# written in a file name "pylint_global.[txt|html]". This option is deprecated
-# and it will be removed in Pylint 2.0.
-files-output=no
```
Comment (romilbhardwaj) [inline .pylintrc:164]: assuming this is removed because its the default

**Review by romilbhardwaj — APPROVED** [review]
Thanks @Michaelvll! Adding to pylint check is a good idea. LGTM.

File: .pylintrc (hunk 1 above)
Comment (Michaelvll) [reply .pylintrc:164]: Yes, this operation is removed, and it seems google's official style guide for python has removed it as well. : )

**Pull request merged**